packages/libllvm/tools-clang-lib-Driver-ToolChains-Linux.cpp.patch

```diff
@@ -1,74 +1,28 @@
+-  addPathIfExists(D, SysRoot + "/usr/lib", Paths);
++  bool nativeBuild = Triple == llvm::Triple(llvm::sys::getDefaultTargetTriple());
++  if (nativeBuild || !IsAndroid)
++    addPathIfExists(D, SysRoot + "/usr/lib", Paths);
 +
-+  if(IsAndroid) {
-+    if (MultiarchTriple == llvm::sys::getDefaultTargetTriple())
-+      addPathIfExists(D, SysRoot + "@TERMUX_PREFIX@/lib", Paths);
-+    else
-+      NativeBuild = false;
-+
-+    if (Arch == llvm::Triple::aarch64) {
-+      addPathIfExists(D, SysRoot + "@TERMUX_PREFIX@/aarch64-linux-android/lib", Paths);
-+      addPathIfExists(D, SysRoot + "/system/lib64", Paths);
-+      if (!NativeBuild)
-+        ExtraOpts.push_back("-rpath=@TERMUX_PREFIX@/aarch64-linux-android/lib");
-+    }
-+    else if (Arch == llvm::Triple::arm || Arch == llvm::Triple::thumb) {
-+      addPathIfExists(D, SysRoot + "@TERMUX_PREFIX@/arm-linux-androideabi/lib", Paths);
-+      addPathIfExists(D, SysRoot + "/system/lib", Paths);
-+      if (!NativeBuild)
-+        ExtraOpts.push_back("-rpath=@TERMUX_PREFIX@/arm-linux-androideabi/lib");
-+    }
-+    else if (Arch == llvm::Triple::x86_64) {
-+      addPathIfExists(D, SysRoot + "@TERMUX_PREFIX@/x86_64-linux-android/lib", Paths);
-+      addPathIfExists(D, SysRoot + "/system/lib64", Paths);
-+      if (!NativeBuild)
-+        ExtraOpts.push_back("-rpath=@TERMUX_PREFIX@/x86_64-linux-android/lib");
-+    }
-+    else if (Arch == llvm::Triple::x86) {
-+      addPathIfExists(D, SysRoot + "@TERMUX_PREFIX@/i686-linux-android/lib", Paths);
-+      addPathIfExists(D, SysRoot + "/system/lib", Paths);
-+      if (!NativeBuild)
-+        ExtraOpts.push_back("-rpath=@TERMUX_PREFIX@/i686-linux-android/lib");
-+    }
-+
-+    ExtraOpts.push_back("-rpath=@TERMUX_PREFIX@/lib");
++  if (IsAndroid) {
++    addPathIfExists(D, SysRoot + "/usr/" + MultiarchTriple + "/lib", Paths);
++    addPathIfExists(D, "/system/" + OSLibDir, Paths);
++    ExtraOpts.push_back("-rpath=" + SysRoot + "/usr/lib");
 +  }
+ }
  
-   // Add the multilib suffixed paths where they are available.
-   if (GCCInstallation.isValid()) {
-@@ -656,8 +691,27 @@
-     return;
- 
-   if (!DriverArgs.hasArg(options::OPT_nostdlibinc))
--    addSystemInclude(DriverArgs, CC1Args, SysRoot + "/usr/local/include");
--
-+	if  (getTriple().isAndroid()) {
-+	  switch (getTriple().getArch()) {
-+        case llvm::Triple::x86_64:
-+	addSystemInclude(DriverArgs, CC1Args, SysRoot + "@TERMUX_PREFIX@/include/x86_64-linux-android");
-+	break;
-+	case llvm::Triple::x86:
-+	addSystemInclude(DriverArgs, CC1Args, SysRoot + "@TERMUX_PREFIX@/include/i686-linux-android");
-+	break;
-+	case llvm::Triple::aarch64:
-+	addSystemInclude(DriverArgs, CC1Args, SysRoot + "@TERMUX_PREFIX@/include/aarch64-linux-android");
-+	break;
-+	case llvm::Triple::arm:
-+	case llvm::Triple::thumb:
-+	addSystemInclude(DriverArgs, CC1Args, SysRoot + "@TERMUX_PREFIX@/include/arm-linux-androideabi");
-+	break;
-+	default:
-+	break;
-+	}
```

Comment:
After setting the default sysroot, [these include directories are all added automatically](https://github.com/llvm/llvm-project/blob/7e77cf473ac9d8f8b65db017d660892f1c8f4b75/clang/lib/Driver/ToolChains/Linux.cpp#L842).